com.intellij.diagnostic.PluginException: Stream closed [Plugin: nl.rubensten.texifyidea]

### Type of JetBrains IDE (IntelliJ, PyCharm, etc.) and version
IntelliJ IDEA 2024.2.3 (build IC-242.23339.11)

### Operating System 
Linux 6.11.3-arch1-1 (amd64)

### TeXiFy IDEA version
0.9.9-alpha.4

### Description



### Stacktrace
```
com.intellij.diagnostic.PluginException: Stream closed [Plugin: nl.rubensten.texifyidea]
	at com.intellij.ide.startup.impl.StartupManagerImplKt$launchActivity$1.invokeSuspend(StartupManagerImpl.kt:502)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith$$$capture(ContinuationImpl.kt:33)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt)
	at --- Async.Stack.Trace --- (captured by IntelliJ IDEA debugger)
	at kotlinx.coroutines.debug.internal.DebugProbesImpl$CoroutineOwner.<init>(DebugProbesImpl.kt:531)
	at kotlinx.coroutines.debug.internal.DebugProbesImpl.createOwner(DebugProbesImpl.kt:510)
	at kotlinx.coroutines.debug.internal.DebugProbesImpl.probeCoroutineCreated$kotlinx_coroutines_core(DebugProbesImpl.kt:497)
	at kotlin.coroutines.jvm.internal.DebugProbesKt.probeCoroutineCreated(DebugProbes.kt:7)
	at kotlin.coroutines.intrinsics.IntrinsicsKt__IntrinsicsJvmKt.createCoroutineUnintercepted(IntrinsicsJvm.kt:161)
	at kotlinx.coroutines.intrinsics.CancellableKt.startCoroutineCancellable(Cancellable.kt:26)
    (...)
	at com.intellij.idea.Main.main(Main.kt:49)
Caused by: java.io.IOException: Stream closed
	at java.base/java.io.BufferedInputStream.getBufIfOpen(BufferedInputStream.java:188)
    (...)
	at kotlin.io.TextStreamsKt.readText(ReadWrite.kt:112)
	at nl.hannahsten.texifyidea.util.CommandRunnerKt$runCommandNonBlocking$2$error$1.invokeSuspend(CommandRunner.kt:55)
	at _COROUTINE._BOUNDARY._(CoroutineDebugging.kt:42)
	at com.intellij.ide.startup.impl.StartupManagerImplKt$launchActivity$1.invokeSuspend(StartupManagerImpl.kt:496)
Caused by: java.io.IOException: Stream closed
	at java.base/java.io.BufferedInputStream.getBufIfOpen(BufferedInputStream.java:188)
    (...)
	at kotlin.io.TextStreamsKt.readText(ReadWrite.kt:112)
	at nl.hannahsten.texifyidea.util.CommandRunnerKt$runCommandNonBlocking$2$error$1.invokeSuspend(CommandRunner.kt:55)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith$$$capture(ContinuationImpl.kt:33)
    (...)
	at kotlinx.coroutines.BuildersKt.async$default(Unknown Source)
	at nl.hannahsten.texifyidea.util.CommandRunnerKt$runCommandNonBlocking$2.invokeSuspend(CommandRunner.kt:55)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith$$$capture(ContinuationImpl.kt:33)
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -3,10 +3,48 @@
 ## [Unreleased]
 
 ### Added
-* Use theme foreground and background colors for equation preview
 
 ### Fixed
+
+## [0.9.9-alpha.9] - 2024-11-29
+
+### Added
+
+* Use theme foreground and background colors for equation preview
+* Add support for \includesvg
+* Support references in \subfix command
+* Provide autocompletion for all relevant directories when using subfiles
+* Do not format existing directories in the given path in the file creation quickfix dialog
+* Disable Evince inverse search on Windows to avoid a UI freeze on opening a second project
+* Use xdg-mime to choose a default pdf viewer in the run configuration template
+* Don't insert the right brace when in front of text
+* Add DeclareMathSymbol to indexed command definitions in installed packages
+* Add support for TEXMFHOME for reference resolving
+* Add diffcoeff and upgreek packages to autocompletion, by @Toseflo
+* Improve reference resolving when using subfiles
+* Add setting to disable auto-import of bibtex entries from remote libraries
+
+### Fixed
+
 * Disallow unmatched braces after \@ifnextchar
+* Fix exception #3771 when a file referenced from cache is deleted
+* Fix basic case of false positive of duplicate label inspection when user defined \if commands are used
+* Fix a parse error when using \else with a user defined \if-command
+* Fix relative path conversion in graphics insertion wizard by resolving relative to the root file
+* Fix exception #3763
+* Fix 'missing import' false positive in subfiles
+* Don't override the file icon for .txt files, by @Steve-Li-1998
+* Fix exceptions #3754 and #3326
+* Fix exceptions in structure view when command parameters are missing
+* Improve error report submitter for long stacktraces
+* Fix a parser issue with bidirectional arrow in TikZ
+* Fix default Docker image name when running Dockerized TeX Live without a project SDK
+* Always use content roots to resolve references
+* Fix 'package not found' error when using texlive-full on Windows, and improve running of system commands, by @tristankretzer
+* Fix rename of files when using subfiles
+* Fix incorrect 'package already included' warning for the subfiles package
+* Fix exception #3557 if using bibtex structure view when bibtex file type is reassignd to plain text
+* Avoid referencing obsolete psifiles, fix exception #3635
 
 ## [0.9.9-alpha.8] - 2024-11-28
 
@@ -558,7 +596,8 @@ Thanks to @jojo2357 and @MisterDeenis for contributing to this release!
 * Fix some intention previews. ([#2796](https://github.com/Hannah-Sten/TeXiFy-IDEA/issues/2796))
 * Other small bug fixes and improvements. ([#2776](https://github.com/Hannah-Sten/TeXiFy-IDEA/issues/2776), [#2774](https://github.com/Hannah-Sten/TeXiFy-IDEA/issues/2774), [#2765](https://github.com/Hannah-Sten/TeXiFy-IDEA/issues/2765)-[#2773](https://github.com/Hannah-Sten/TeXiFy-IDEA/issues/2773))
 
-[Unreleased]: https://github.com/Hannah-Sten/TeXiFy-IDEA/compare/v0.9.9-alpha.8...HEAD
+[Unreleased]: https://github.com/Hannah-Sten/TeXiFy-IDEA/compare/v0.9.9-alpha.9...HEAD
+[0.9.9-alpha.9]: https://github.com/Hannah-Sten/TeXiFy-IDEA/compare/v0.9.9-alpha.8...v0.9.9-alpha.9
 [0.9.9-alpha.8]: https://github.com/Hannah-Sten/TeXiFy-IDEA/compare/v0.9.9-alpha.7...v0.9.9-alpha.8
 [0.9.9-alpha.7]: https://github.com/Hannah-Sten/TeXiFy-IDEA/compare/v0.9.9-alpha.6...v0.9.9-alpha.7
 [0.9.9-alpha.6]: https://github.com/Hannah-Sten/TeXiFy-IDEA/compare/v0.9.9-alpha.4...v0.9.9-alpha.6

--- a/gradle.properties
+++ b/gradle.properties
@@ -1,4 +1,4 @@
-pluginVersion = 0.9.9-alpha.8
+pluginVersion = 0.9.9-alpha.9
 
 #     Info about build ranges: https://www.jetbrains.org/intellij/sdk/docs/basics/getting_started/build_number_ranges.html
 #    Note that an xyz branch corresponds to version 20xy.z and a since build of xyz.*

--- a/src/nl/hannahsten/texifyidea/util/CommandRunner.kt
+++ b/src/nl/hannahsten/texifyidea/util/CommandRunner.kt
@@ -52,7 +52,18 @@ suspend fun runCommandNonBlocking(
 
         process.outputWriter().use { if (input != null) it.write(input) }
         val output = if (!discardOutput) async { process.inputReader().use { it.readText() } } else null
-        val error = if (!discardOutput) async { process.errorReader().use { it.readText() } } else null
+        val error = if (!discardOutput) async { process.errorReader().use {
+            try {
+                it.readText()
+            } catch (e: IOException) {
+                // In some case directly after IDE start, the stream may be closed already, so ignore that
+                if (e.message?.contains("Stream closed") == true) {
+                    Log.info("Ignored closed stream: " + e.message)
+                    e.message
+                }
+                else throw e
+            }
+        } } else null
 
         withTimeoutOrNull(1_000 * timeout) {
             process.awaitExit()

--- a/src/nl/hannahsten/texifyidea/util/CommandRunner.kt
+++ b/src/nl/hannahsten/texifyidea/util/CommandRunner.kt
@@ -52,18 +52,24 @@ suspend fun runCommandNonBlocking(
 
         process.outputWriter().use { if (input != null) it.write(input) }
         val output = if (!discardOutput) async { process.inputReader().use { it.readText() } } else null
-        val error = if (!discardOutput) async { process.errorReader().use {
-            try {
-                it.readText()
-            } catch (e: IOException) {
-                // In some case directly after IDE start, the stream may be closed already, so ignore that
-                if (e.message?.contains("Stream closed") == true) {
-                    Log.info("Ignored closed stream: " + e.message)
-                    e.message
+        val error = if (!discardOutput) {
+            async {
+                process.errorReader().use {
+                    try {
+                        it.readText()
+                    }
+                    catch (e: IOException) {
+                        // In some case directly after IDE start, the stream may be closed already, so ignore that
+                        if (e.message?.contains("Stream closed") == true) {
+                            Log.info("Ignored closed stream: " + e.message)
+                            e.message
+                        }
+                        else throw e
+                    }
                 }
-                else throw e
             }
-        } } else null
+        }
+        else null
 
         withTimeoutOrNull(1_000 * timeout) {
             process.awaitExit()

